iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/bicep/ParameterDeclarationImplTest.java

```diff
@@ -93,4 +95,58 @@ void shouldParseParameterDeclarationForResource() {
     assertThat(tree.resourceType().value()).isEqualTo("myResource");
     assertThat(tree.children()).map(token -> ((TextTree) token).value()).containsExactly("param", "myParam", "resource", "myResource");
   }
+
+  @Test
+  void shouldParseIntParameterDeclarationWithDecorator() {
+    String code = code(
+      "@description('parameter description')",
+      "@minValue(0)",
+      "@maxValue(10)",
+      "param myParam int");
+    ParameterDeclaration tree = parse(code, BicepLexicalGrammar.PARAMETER_DECLARATION);
```

Comment:
I would add another test case where `@minValue('0')`.
Even if we shouldn't care of invalid code, I would document how the parser react to it (do we crash? throw an exception?)

Reply:
The current implementation would fail with `ClassCastException`. Added corresponding tests.